features/environment.py

```diff
@@ -15,16 +19,46 @@ def before_all(_):
 def before_scenario(context, _):
     context.test_start_local_datetime = datetime.now()
     context.action_plan_id = str(uuid.uuid4())
-    _purge_queues()
+    _clear_down_all_queues()
 
 
-def _purge_queues():
-    with RabbitContext() as rabbit:
-        rabbit.channel.queue_purge(queue=Config.RABBITMQ_RH_OUTBOUND_CASE_QUEUE)
-        rabbit.channel.queue_purge(queue=Config.RABBITMQ_RH_OUTBOUND_UAC_QUEUE)
-        rabbit.channel.queue_purge(queue=Config.RABBITMQ_SAMPLE_INBOUND_QUEUE)
-        rabbit.channel.queue_purge(queue=Config.RABBITMQ_OUTBOUND_FIELD_QUEUE)
-        rabbit.channel.queue_purge(queue=Config.RABBITMQ_UNADDRESSED_REQUEST_QUEUE)
+def get_msg_count(queue_name):
+    uri = f'http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_MAN_PORT}/api/queues/%2f/{queue_name}'
+    response = requests.get(uri, auth=(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
+    response.raise_for_status()
+    response_data = json.loads(response.content)
+
+    return response_data['messages']
+
+
+def _clear_down_all_queues():
+    all_queues = _get_all_queues()
+
+    for queue in all_queues:
+        # keep killing this Delayed queue, just to stop it redlivering anything in some mad race condition
+        _clear_down_queue(Config.RABBITMQ_DELAYED_REDELIVERY_QUEUE)
+        _clear_down_queue(queue)
+
+
+def _get_all_queues():
+    uri = f'http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_MAN_PORT}/api/queues/%2f/'
+    response = requests.get(uri, auth=(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
+    response.raise_for_status()
+    response_data = json.loads(response.content)
+
+    return [queue['name'] for queue in response_data]
+
+
+def _clear_down_queue(queue_name):
+    while True:
+        uri = f'http://{Config.RABBITMQ_HOST}:{Config.RABBITMQ_MAN_PORT}/api/queues/%2f/{queue_name}/contents'
+        response = requests.delete(uri, auth=(Config.RABBITMQ_USER, Config.RABBITMQ_PASSWORD))
```

Comment:
Is it worth calling `get_msg_count` first and then make the delete request if messages > 0 afterwards? Might save a few API calls this way? Probably won't make a huge difference though